connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2092,6 +2097,11 @@ class SparkConnectPlanner(val session: SparkSession) {
       case path => writer.start(path)
     }
 
+    // Register the new query so that the session and query references are cached.
+    SparkConnectService.streamingSessionManager.registerNewStreamingQuery(
```

Comment:
If the query throws before this line. What would happen? I would think the `orElse` would be triggered:
```
val query = SparkConnectService.streamingSessionManager
      .findCachedQuery(id, session) // Common case: query is cached in connect session manager.
      .orElse { // Else try to find it in active streams. Mostly will not be found here.
        Option(session.streams.get(id))
```
And that'd still return a query ID not found exception?

Reply:
Ah I see then the query won't be returned so it's fine

Reply:
Yep, query id is not sent to the client, so no need to track.